src/charm.py

```diff
@@ -118,6 +118,18 @@ def calculate_env(self) -> jenkins.Environment:
             JENKINS_PREFIX=self.ingress_observer.get_path(),
         )
 
+    def _is_ingress_correctly_configured(self) -> typing.Tuple[bool, str]:
+        """Indicate whether ingress is properly configured.
+
+        Returns:
+            A tuple of a bool indicating whether ingress is configured and the reason if not.
```

Comment:
What about making this a named tuple so that we don't have to assign meaning to indexes of the tuple implicitly?